label_studio/io_storages/base_models.py

```diff
@@ -76,12 +76,25 @@ def can_resolve_url(self, url):
 
     def resolve_uri(self, uri):
         try:
-            extracted_uri, extracted_storage = get_uri_via_regex(uri, prefixes=(self.url_scheme,))
-            if not extracted_storage:
-                logger.info(f'No storage info found for URI={uri}')
-                return
-            http_url = self.generate_http_url(extracted_uri)
-            return uri.replace(extracted_uri, http_url)
+            if type(uri) == str:
+                uri = [uri]
+
+            resolved_uris = []
+            for sub_uri in uri:
+                extracted_uri, extracted_storage = get_uri_via_regex(
+                    sub_uri, prefixes=(self.url_scheme,)
+                )
+                if not extracted_storage:
+                    logger.info(f'No storage info found for URI={sub_uri}')
+                    return
+
+                http_url = self.generate_http_url(extracted_uri)
+                resolved_uris.append(sub_uri.replace(extracted_uri, http_url))
+
+            if len(uri) == 1:
```

Comment:
what is task.data = 
```
{ images: ["1.jpg"] }
```
this core will incorrectly return resolved_uris[0], however it must be resovled_uris. 

I would write something like this: 

```
def resolve_uri(self, uri):
  # process list of urls
  if isinstance(uri, list):
      resolved = []
      for sub in url:
          resolved.append(self.resolve_uri(sub))
      return resolved

  # process one url
  else:
        try:
            extracted_uri, extracted_storage = get_uri_via_regex(uri, prefixes=(self.url_scheme,))
            if not extracted_storage:
                logger.info(f'No storage info found for URI={uri}')
                return
            http_url = self.generate_http_url(extracted_uri)
            return uri.replace(extracted_uri, http_url)
        except Exception as exc:
            logger.info(f'Can\'t resolve URI={uri}', exc_info=True)

Reply:
Thanks for the feedback. I agree with your suggestion.

Reply:
Thank you for your changes. Now we should wait for tests from our QA team @borisheartex